plugins/rebalance/transaction.go

```diff
@@ -0,0 +1,257 @@
+        reqDict["rebalance-command"] =  command
+
+log.Info ("Stopping rebalance : command = ", command);
+
+        req := &brick.GfBrickOpReq{
+                Name: xlator_name,
+                Op:   int(brick.OpBrickXlatorDefrag),
+        }
+
+        req.Input, err = dict.Serialize(reqDict)
+        if err != nil {
+                c.Logger().WithError(err).WithField(
+                        "volume", volname).Error("failed to serialize dict for rebalance stop")
+        }
+
+        var rsp brick.GfBrickOpRsp
+        err = client.Call("Brick.OpBrickXlatorDefrag", req, &rsp)
+        if err != nil || rsp.OpRet != 0 {
+                c.Logger().WithError(err).WithField(
+                        "volume", volname).Error("failed to send rebalance stop RPC")
+                return err
+        }
+
+
+        // TODO : Send a response
+        // Unserialize the resp dict for rebalance stop
+     //   rspDict, err := dict.Unserialize(rsp.Output)
+    //    if err != nil {
+   //             log.WithError(err).Error("dict unserialize failed")
+  //              return err
+ //       }
+
+        return nil
+}
+
+
+
+func txnRebalanceStatus(c transaction.TxnCtx) error {
+
+	var rebalinfo rebalanceapi.RebalInfo
+        var rebalNodeStatus rebalanceapi.RebalNodeStatus
+
+	err := c.Get("rinfo", &rebalinfo)
+	if err != nil {
+		return err
+	}
+
+        var volname string
+        var xlator_name string
+        var command string
+
+        if err := c.Get("volname", &volname); err != nil {
+                c.Logger().WithError(err).WithField(
+                        "key", "volname").Error("failed to get value for key from context")
+                return err
+        }
+
+        if rebalinfo.Status !=  rebalanceapi.Started {
+                //Rebalance has completed. Send the stored information
+
+                rebalNodeStatus = rebalinfo.RebalStats
+                c.SetNodeResult(gdctx.MyUUID, rebalStatusTxnKey, rebalNodeStatus)
+                return nil
+
+        }
+
+        // What is the expected behaviour if the process does not exist (rebalance has completed)?
+        // Will it restart the process?
+
+	rebalanceProcess, err := NewRebalanceProcess(rebalinfo)
+	if err != nil {
+
+                // TODO: Send the stored Info?
+		return err
+	}
+
+        client, err := daemon.GetRPCClient(rebalanceProcess)
+        if err != nil {
+                c.Logger().WithError(err).WithField(
+                        "volume", volname).Error("failed to connect to the rebalance process")
+
+                // Get status from store
+                rebal, err := GetRebalanceInfo(volname)
+	        if err != nil {
+		        return err
+	        }
+
+                rebalNodeStatus = rebal.RebalStats
+
+        } else {
+
+                //Send the status request
+
+                reqDict := make(map[string]string)
+
+                xlator_name = fmt.Sprintf("%s-distribute", volname)
+
+                // TODO: FIX THIS!!!
```

Comment:
what fix is required here?

Reply:
I wasn't sure if this was the best way to convert a const uint64 to a string (this was the only way I got it to work)

Reply:
`strconv.FormatUint` Ref: https://golang.org/pkg/strconv/

Reply:
I tried that but it didn't work:
plugins/rebalance/transaction.go:88:51: cannot use rebalance.CmdStop (type rebalance.Command) as type uint64 in argument to strconv.FormatUint